packages/wasm-sdk/src/dpp.rs

```diff
@@ -295,11 +295,18 @@ impl IdentityWasm {
 // }
 
 #[wasm_bindgen]
-pub struct DataContractWasm(DataContract);
+pub struct DataContractWasm(DataContract, &'static PlatformVersion);
+
+impl DataContractWasm {
+    pub fn new(data_contract: DataContract, platform_version: &'static PlatformVersion) -> Self {
+        Self(data_contract, platform_version)
+    }
+}
 
 impl From<DataContract> for DataContractWasm {
     fn from(value: DataContract) -> Self {
-        Self(value)
+        // Use first version as fallback for backward compatibility
+        Self(value, PlatformVersion::first())
```

Comment:
I don't think this is a safe approach. For now, I would just pass platfrom version to all methods that require it (as rust dpp does). There are non-instance methods and you would need to do it anyway.

Reply:
Pushed some updates, but I see you've been working on dpp.rs in #2791 so maybe I will pause (since I don't exactly know what I'm doing here :zany_face:)